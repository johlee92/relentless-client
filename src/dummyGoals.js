export default {
    "annual_goals": [
        {
            "id": "1",
            "content": "Improve my physical, mental and spiritual aspects of health",
            "modified": "2020-01-01T00:00:00.000Z",
            "complete": "false"
        },
        {
            "id": "2",
            "content": "To develop confidence in personal finance management, including increase savings rate to at least 30% of post-tax income",
            "modified": "2020-01-01T00:00:00.000Z",
            "complete": "false"
        },
        {
            "id": "3",
            "content": "Seeing how my career trajectory fit in overall healthcare industry by connecting myself to resources {communication skill-building, finding mentor}",
            "modified": "2020-01-01T00:00:00.000Z",
            "complete": "false"
        }
    ],
    "monthly_goals": [
        {
            "id": "1",
            "content": "Continue to build on my exercise routine. Be able to swim 800m freestyle in under 10 minutes",
            "modified": "2020-02-01T00:00:00.000Z",
            "annual_goal_id":"1",
            "monthNum": "2",
            "complete": "false"
        },
        {
            "id": "2",
            "content": "Review past several month’s spending, identify areas of cost savings, and act on those cost savings areas",
            "modified": "2020-02-01T00:00:00.000Z",
            "annual_goal_id":"2",
            "monthNum": "2",
            "complete": "false"
        },
        {
            "id": "3",
            "content": "Study reports of pharmaceutical companies and consulting groups",
            "modified": "2020-02-01T00:00:00.000Z",
            "annual_goal_id":"3",
            "monthNum": "2",
            "complete": "false"
        }
    ],
    "weekly_goals": [
        {
            "id": "1",
            "content": "Part 1. Continue to build on my exercise routine. Be able to swim 800m freestyle in under 10 minutes",
            "modified": "2020-02-13T00:00:00.000Z",
            "monthly_goal_id":"1",
            "weekNum": "8",
            "complete": "true"
        },
        {
            "id": "2",
            "content": "Part 1. Review past several month’s spending, identify areas of cost savings, and act on those cost savings areas",
            "modified": "2020-02-13T00:00:00.000Z",
            "monthly_goal_id":"2",
            "weekNum": "8",
            "complete": "false"
        },
        {
            "id": "3",
            "content": "Part 1. Study reports of pharmaceutical companies and consulting groups",
            "modified": "2020-02-13T00:00:00.000Z",
            "monthly_goal_id":"3",
            "weekNum": "8",
            "complete": "false"
        },
        // {
        //     "id": "4",
        //     "content": "Part 2. Continue to build on my exercise routine. Be able to swim 800m freestyle in under 10 minutes",
        //     "modified": "2020-02-20T00:00:00.000Z",
        //     "monthly_goal_id":"1",
        //     "weekNum": "9",
        //     "complete": "true"
        // },
        // {
        //     "id": "5",
        //     "content": "Part 2. Review past several month’s spending, identify areas of cost savings, and act on those cost savings areas",
        //     "modified": "2020-02-20T00:00:00.000Z",
        //     "monthly_goal_id":"2",
        //     "weekNum": "9",
        //     "complete": "true"
        // },
        // {
        //     "id": "6",
        //     "content": "Part 2. Study reports of pharmaceutical companies and consulting groups",
        //     "modified": "2020-02-20T00:00:00.000Z",
        //     "monthly_goal_id":"3",
        //     "weekNum": "9",
        //     "complete": "true"
        // },
        // {
        //     "id": "7",
        //     "content": "Part 3. Continue to build on my exercise routine. Be able to swim 800m freestyle in under 10 minutes",
        //     "modified": "2020-02-01T00:00:00.000Z",
        //     "monthly_goal_id":"1",
        //     "weekNum": "10",
        //     "complete": "true"
        // },
        // {
        //     "id": "8",
        //     "content": "Part 3. Review past several month’s spending, identify areas of cost savings, and act on those cost savings areas",
        //     "modified": "2020-02-01T00:00:00.000Z",
        //     "monthly_goal_id":"2",
        //     "weekNum": "10",
        //     "complete": "true"
        // },
        // {
        //     "id": "9",
        //     "content": "Part 3. Study reports of pharmaceutical companies and consulting groups",
        //     "modified": "2020-02-01T00:00:00.000Z",
        //     "monthly_goal_id":"3",
        //     "weekNum": "10",
        //     "complete": "false"
        // }
    ],
  }
  