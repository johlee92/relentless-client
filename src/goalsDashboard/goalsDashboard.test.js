import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GoalsDashboard from './goalsDashboard';
import { BrowserRouter } from 'react-router-dom';
import dummyGoals from '../dummyGoals'

describe.only(`GoalsDashboard component`, () => {
    const props = {
        className: 'test-class-name',
        viewGoals: 'annual',
        currentWeekGoals: dummyGoals.weekly_goals,
        currentGoals: {
            weeklyGoals: dummyGoals.weekly_goals,
            monthlyGoals: dummyGoals.monthly_goals,
            annualGoals: dummyGoals.annual_goals
        },
        annual_goals: dummyGoals.annual_goals,
        monthly_goals: dummyGoals.monthly_goals,
        weekly_goals: dummyGoals.weekly_goals
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <GoalsDashboard {...props}/>
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component from props', () => {
        const wrapper = shallow(<GoalsDashboard {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
