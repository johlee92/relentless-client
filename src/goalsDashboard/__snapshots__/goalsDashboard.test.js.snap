// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsDashboard component renders the component from props 1`] = `
<span>
  <div
    className="dashboard"
  >
    <div />
  </div>
  <div
    className="dashboard"
  >
    <div
      className="dashboard-cards"
    >
      Weekly Goals Completion: 
      3/3
    </div>
    <div
      className="dashboard-cards"
    >
      YTD Accomplishment: 
      9/9
    </div>
  </div>
</span>
`;
